src/LazyBulderCommentsStatistics.php

```diff
@@ -26,7 +27,7 @@ class LazyBulderCommentsStatistics {
   /**
    * Constructs a new LazyBuilders object.
    */
-  public function __construct(EntityManagerInterface $entity_manager, CommentStatisticsInterface $comment_statistics) {
+  public function __construct(EntityTypeManagerInterface $entity_manager, CommentStatisticsInterface $comment_statistics) {
```

Comment:
When are you creating object of this class ? 
I think It has link to https://github.com/wearewondrous/extra_fields_collection/blob/d9-compatibility/src/ExtraFieldsCollectionServiceProvider.php#L32
And this entity.manager should be entity_type.manager.

Reply:
This class is never instantiated. Here, in https://github.com/wearewondrous/extra_fields_collection/blob/d9-compatibility/src/ExtraFieldsCollectionServiceProvider.php#L32, the Reference class takes a string as a param which references a service. I have updated it.